pandas-stubs/core/frame.pyi

```diff
@@ -698,16 +698,44 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
         self,
         expr: _str,
         *,
+        parser: Literal["pandas", "python"] = ...,
+        engine: Literal["python", "numexpr"] | None = ...,
+        local_dict: dict[_str, Any] | None = ...,
+        global_dict: dict[_str, Any] | None = ...,
+        resolvers: list[Mapping] | None = ...,
+        level: int = ...,
+        target: object | None = ...,
         inplace: Literal[True],
-        **kwargs: Any,  # TODO: make more precise https://github.com/pandas-dev/pandas-stubs/issues/1173
+    ) -> None: ...
+    @overload
+    def query(
+        self,
+        expr: _str,
+        *,
+        inplace: Literal[True],
+        **kwargs: Any,
     ) -> None: ...
     @overload
     def query(
         self,
         expr: _str,
         *,
         inplace: Literal[False] = ...,
-        **kwargs: Any,  # TODO: make more precise https://github.com/pandas-dev/pandas-stubs/issues/1173
+        parser: Literal["pandas", "python"] = ...,
+        engine: Literal["python", "numexpr"] | None = ...,
+        local_dict: dict[_str, Any] | None = ...,
+        global_dict: dict[_str, Any] | None = ...,
+        resolvers: list[Mapping] | None = ...,
+        level: int = ...,
+        target: object | None = ...,
+    ) -> Self: ...
+    @overload
+    def query(
+        self,
+        expr: _str,
+        *,
+        inplace: Literal[False] = ...,
+        **kwargs: Any,
     ) -> Self: ...
```

Comment:
same comment - not sure why we need this overload when the previous one covers it.

Reply:
The goal was to still allow for someone passing kwargs as a dictionary instead of the individual arguments since it is the way documented in the docs like `df.query("col1 > col2", **kwargs)` that you would pass from another function.

Reply:
The idea is that someone may put a wrapper like: 
```python
def my_own_query(df, expr, **kwargs):
    return df.query(expr, **kwargs)
```
If you drop the second overload this will raise an error.

Reply:
I think the goal here is to restrict the possible keyword arguments, so we should remove the overload.